rln-research/membership-merkle-tree-storage-evaluation.md

```diff
@@ -0,0 +1,107 @@
+# Waku-RLN-Relay: Evaluating Storage Overhead of Membership Merkle Tree   
+## Introduction
+
+In the current design of the Waku-RLN-Relay protocol, the membership Merkle tree (MT) is locally persisted by each peer. This incurs some storage overhead which may or may not fit resource-limited devices. This overhead also depends on the algorithm used to represent the Merkle tree. So far, we are aware of two algorithms namely, Full Merkle tree (FMT) and Optimal Merkle tree (OMT), and in this post, we would like
```

Comment:
What is "Optimal" MT? I've only heard of incremental MT algorithm. Is this referring to the same or something different?

Reply:
This is just a naming for the MT implementation of killic lib, tbh, I have no string opinion about its naming. Though, I am sure it differs from Incremental Merkle tree and also is not called sparse Merkle tree. We can alternatively name it `Indexed Merkle Tree` as each node in the tree is stored alongside its index in the tree. Will search further to see if there is a specific name for this algorithm.